core/citrus-api/src/main/java/org/citrusframework/report/TestResults.java

```diff
@@ -35,15 +39,24 @@ public class TestResults {
 
     private static final long serialVersionUID = 1L;
 
-    /** Common decimal format for percentage calculation in report **/
+    private static final String DURATION_KEY = "duration";
+
+    /**
+     * Common decimal format for percentage calculation in report
+     **/
     private static final DecimalFormat decFormat = new DecimalFormat("0.0");
     private static final String ZERO_PERCENTAGE = "0.0";
 
-    /** Collected test results */
-    private final Set<TestResult> results = Collections.synchronizedSet(new LinkedHashSet<>());
+    /**
+     * Collected test results
+     */
+    private final Set<TestResult> results = synchronizedSet(new LinkedHashSet<>());
+
+    private final Map<String, Duration> totalDuration = new ConcurrentHashMap<>(1);
```

Comment:
for synchronization purposes. it's a `ConcurrentHashMap`. I've mingled with different ideas, but finally ended up here. I think, that this is the easiest one.

Reply:
Understood. Of course there a plenty of ways to sync this. 
In my opinion, storing the duration in millis in an AtomicLong would be a straight forward and a little less confusing, it would also avoid the creation of new Duration objects, thereby reducing memory overhead. 

If you want to stay with duration, you may want to consider switching to an AtomicReference<Duration>:

```java
totalDuration.accumulateAndGet(result.getDuration(), Duration::plus);
```

Reply:
> it would also avoid the creation of new Duration objects, thereby reducing memory overhead.

what do you mean by this? I don't see the `Duration` being re-initialized 😲 

the downside of your suggestion is of course, that only milliseconds are available after. the current way we can expose the `Duration` and leave formatting etc. up to the implementation.

Reply:
Duration is immutable. So every add creates a new Object, leaving the old one for the garbage collector. Quite accademical in this context. 

If you want to expose the Duration, do it with:
```java
return Duration.ofMillis(totalDuration);
```
That  also creates a new object but it is probably just called once.

Alternatively, as poposed exchanges your Map by an AtomicReference and stick with Duration.

Reply:
thanks for the explanation. I find this reasonable 👍🏼

Reply:
It was a pleasure